core/MyBlob.cpp

```diff
@@ -0,0 +1,221 @@
+/*
+ * The MySensors Arduino library handles the wireless radio link and protocol
+ * between your home built sensors/actuators and HA controller of choice.
+ * The sensors forms a self healing radio network with optional repeaters. Each
+ * repeater and gateway builds a routing tables in EEPROM which keeps track of the
+ * network topology allowing messages to be routed to nodes.
+ *
+ * Created by Constnatin Petra <constantin.petra@gmail.com>
+ * Copyright (C) 2022 Constantin Petra
+ * Full contributor list: https://github.com/mysensors/MySensors/graphs/contributors
+ *
+ * Documentation: http://www.mysensors.org
+ * Support Forum: http://forum.mysensors.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * version 2 as published by the Free Software Foundation.
+ */
+
+#include "Version.h"
+#include "MyConfig.h"
+#include "MyEepromAddresses.h"
+#include "MyMessage.h"
+#include "MyBlob.h"
+#include "stddef.h"
+
+#define SET_PAYLOAD_TYPE(u, type)  BF_SET(u, type, V2_MYS_HEADER_CEP_PAYLOADTYPE_POS, \
+        V2_MYS_HEADER_CEP_PAYLOADTYPE_SIZE)
+#define GET_PAYLOAD_TYPE(u)        BF_GET(u, V2_MYS_HEADER_CEP_PAYLOADTYPE_POS, \
+        V2_MYS_HEADER_CEP_PAYLOADTYPE_SIZE)
+#define SET_COMMAND(u, command)    BF_SET(u, command, V2_MYS_HEADER_CEP_COMMAND_POS, \
+        V2_MYS_HEADER_CEP_COMMAND_SIZE)
+#define GET_COMMAND(u)             BF_GET(u, V2_MYS_HEADER_CEP_COMMAND_POS, \
+        V2_MYS_HEADER_CEP_COMMAND_SIZE)
+#define SET_LENGTH(u, finalLength) BF_SET(u, finalLength, V2_MYS_HEADER_VSL_LENGTH_POS, \
+        V2_MYS_HEADER_VSL_LENGTH_SIZE)
+#define GET_LENGTH(u)              BF_GET(u, V2_MYS_HEADER_VSL_LENGTH_POS, \
+        V2_MYS_HEADER_VSL_LENGTH_SIZE)
+#define SET_VERSION(u)             BF_SET(u, V2_MYS_HEADER_PROTOCOL_VERSION, \
+        V2_MYS_HEADER_VSL_VERSION_POS, V2_MYS_HEADER_VSL_VERSION_SIZE);
+
+#define MAX_BLOB_SIZE (MAX_PAYLOAD_SIZE)
+
+// replicates part of the MyMesgage structure related to a single sensor data.
+typedef struct {
+	uint8_t command_echo_payload;
+	uint8_t type;
+	uint8_t sensor;
+	union {
+		uint8_t bValue;
+		uint16_t uiValue;
+		int16_t iValue;
+		uint32_t ulValue;
+		int32_t lValue;
+		struct {
+			float fValue;
+			uint8_t fPrecision;
+		};
+	} __attribute__((packed));
+} __attribute__((packed)) blob;
+
+MyBlob::~MyBlob()
```

Comment:
Perhaps renaming MyBlob to MyMultiMessage, to keep naming more similar to MyMessage?

Reply:
I was kind of afraid of that :) I am actually quite fond of the "Blob" naming, in my programming history it always meant a conglomerate of different type of data. I also find it fun to use and easy to remember. There is, now, the V_MULTI_MESSAGE sensor type that makes the connection for the unaware.